tests/ResponseTest.php

```diff
@@ -160,4 +160,27 @@ public function toArray()
         $this->assertSame('/user/123', $page->url);
         $this->assertSame('123', $page->version);
     }
+
+    public function test_xhr_partial_response()
+    {
+        $request = Request::create('/user/123', 'GET');
+        $request->headers->add(['X-Inertia' => 'true']);
+        $request->headers->add(['X-Inertia-Partial-Component' => 'User/Edit']);
+        $request->headers->add(['X-Inertia-Partial-Data' => 'partial']);
+
+        $user = (object) ['name' => 'Jonathan'];
+        $response = new Response('User/Edit', ['user' => $user, 'partial' => 'partial-data'], 'app', '123');
+        $response = $response->toResponse($request);
+        $page = $response->getData();
+
+        $props = get_object_vars($page->props);
+
+        $this->assertInstanceOf(JsonResponse::class, $response);
+        $this->assertSame('User/Edit', $page->component);
+        $this->assertFalse(isset($props['name']));
```

Comment:
Shouldn’t you check for the user key here?

Reply:
The user key is not part of the partial data, the only key present is the `partial` key. See Line 169

Reply:
Yeah, that's why you want to check it is not there right? The `name` prop is within the `user` right?

```diff
- $this->assertFalse(isset($props['name']));
+ $this->assertFalse(isset($props['user']));

Reply:
Ah! I see what you mean now. Good catch!

I'll update this after work today. 😃